File: models/ambulance/AmbulanceAssignment.js
// models/ambulance/AmbulanceAssignment.js
import mongoose from "mongoose";

const AmbulanceAssignmentSchema = new mongoose.Schema({
  hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
  branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

  ambulanceId: { type: mongoose.Schema.Types.ObjectId, ref: "AmbulanceMaster", required: true },
  driverId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },

  startTime: { type: Date, required: true },
  endTime: Date,
  status: { type: String, enum: ["Active", "Completed"], default: "Active" },

  createdBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" }
}, { timestamps: true });

export default mongoose.model("AmbulanceAssignment", AmbulanceAssignmentSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/AmbulanceAssignment.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "ambulance_assignments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AmbulanceAssignment extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ambulance_id", nullable = false)
    private AmbulanceMaster ambulance;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "driver_id", nullable = false)
    private User driver;

    @Column(name = "start_time", nullable = false)
    private LocalDateTime startTime;

    @Column(name = "end_time")
    private LocalDateTime endTime;

    @Enumerated(EnumType.STRING)
    private Status status = Status.Active;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    public enum Status {
        Active, Completed
    }
}
